### PR TITLE
update mod_pubsub docs to reflect current state [skip ci]

### PR DESCRIPTION
**Pull request by michalwski**
This PR updates `mod_pubsub`s documentation.

### CODE REVIEW

File: doc/modules/mod_pubsub.md [hunk 1]
```diff
@@ -65,16 +65,8 @@ Only one nodetree can be used per host and is shared by all node plugins.
...
-
-#### `<<"virtual">>`
-
-Does not store nodes in a database.
-This saves resources on systems when the number of nodes is large.
-When using the `<<"virtual">>` nodetree, you can only enable these node plugins: `[<<“flat”>>,<<“pep”>>]` or `[<<“flat”>>]`; any other plugin configuration will not work.
-Also, all nodes will have the defaut configuration, and this can not be changed.
-
-Using the `<<“virtual”>>` nodetree requires the pubsub module to be started with a clean database, it will not work if you used the default `<<“tree”>>` nodetree before.
+Stores nodes in a database (`mnesia` and `rdbms`).
```
Comment (aleklisi) [inline doc/modules/mod_pubsub.md]: Stores nodes in a database (`mnesia` and `rdbms`).
or 
Stores nodes in a database (`mnesia` or `rdbms`).
??

File: doc/modules/mod_pubsub.md [hunk 2]
```diff
@@ -124,12 +116,26 @@ If you'd like to learn more about metrics in MongooseIM, please visit the [Mongo
 
 | Name | Type | Description (when it gets incremented) |
 | ---- | ---- | -------------------------------------- |
-| `[global, backends, mod_pubsub_db, set_state]` | histogram | Time to update user's state for specific node. |
+| `[global, backends, mod_pubsub_db, set_state] | histogram | Time to update user's state for specific node. |
```
Comment (aleklisi) [inline doc/modules/mod_pubsub.md]: is this intended change?

File: doc/modules/mod_pubsub.md (hunk 1 above)
Comment (aleklisi) [reply doc/modules/mod_pubsub.md]: @goddammit ??

File: doc/modules/mod_pubsub.md (hunk 2 above)
Comment (michalwski) [reply doc/modules/mod_pubsub.md]: That's accidental, good catch!

File: doc/modules/mod_pubsub.md (hunk 1 above)
Comment (goddammit) [reply doc/modules/mod_pubsub.md]: Or 
"Stores nodes in a database (either `mnesia` or `rdbms`)" if I understand you correctly

File: doc/modules/mod_pubsub.md [hunk 3]
```diff
@@ -65,16 +65,8 @@ Only one nodetree can be used per host and is shared by all node plugins.
...
-#### `<<"virtual">>`
-
-Does not store nodes in a database.
-This saves resources on systems when the number of nodes is large.
-When using the `<<"virtual">>` nodetree, you can only enable these node plugins: `[<<“flat”>>,<<“pep”>>]` or `[<<“flat”>>]`; any other plugin configuration will not work.
-Also, all nodes will have the defaut configuration, and this can not be changed.
-
-Using the `<<“virtual”>>` nodetree requires the pubsub module to be started with a clean database, it will not work if you used the default `<<“tree”>>` nodetree before.
+Stores nodes in a database (`mnesia` and `rdbms`).
+Allows to organise nodes in tries if `hometree` plugin is used.
```
Comment (goddammit) [inline doc/modules/mod_pubsub.md]: Allows to organise nodes in trees if the `hometree` plugin is used.

File: doc/modules/mod_pubsub.md [hunk 4]
```diff
@@ -132,4 +124,18 @@ If you'd like to learn more about metrics in MongooseIM, please visit the [Mongo
 | `[global, backends, mod_pubsub_db, get_states_by_bare]` | histogram | Time to fetch nodes' states for bare JID. |
 | `[global, backends, mod_pubsub_db, get_states_by_full]` | histogram | Time to fetch nodes' states for full JID. |
 | `[global, backends, mod_pubsub_db, get_own_nodes_states]` | histogram | Time to fetch state data for user's nodes. |
-
+| `[global, backends, mod_pubsub_db, create_node, ]` | histogram | Time to set owner after node is created. |
```
Comment (goddammit) [inline doc/modules/mod_pubsub.md]: Time to set the owner after a node is created.

File: doc/modules/mod_pubsub.md [hunk 5]
```diff
@@ -132,4 +124,18 @@ If you'd like to learn more about metrics in MongooseIM, please visit the [Mongo
...
+| `[global, backends, mod_pubsub_db, del_node]` | histogram | Time to delete all data related to a node being removed. |
+| `[global, backends, mod_pubsub_db, get_items]` | histogram | Time to fetch node's items. |
+| `[global, backends, mod_pubsub_db, get_item]` | histogram | Time to fetch a specific item for a node. |
+| `[global, backends, mod_pubsub_db, set_item]` | histogram | Time to insert/update item in a node. |
+| `[global, backends, mod_pubsub_db, del_item]` | histogram | Time to delete an item from a node. |
+| `[global, backends, mod_pubsub_db, del_items]` | histogram | Time to delete specified items from a node |
```
Comment (goddammit) [inline doc/modules/mod_pubsub.md]: Time to delete specified items from a node.

File: doc/modules/mod_pubsub.md [hunk 6]
```diff
@@ -132,4 +124,18 @@ If you'd like to learn more about metrics in MongooseIM, please visit the [Mongo
...
+| `[global, backends, mod_pubsub_db, set_node]` | histogram | Time to create/update a node. |
+| `[global, backends, mod_pubsub_db, find_node_by_id,]` | histogram | Time to fetch a node by its id. |
+| `[global, backends, mod_pubsub_db, find_nodes_by_key]` | histogram | Time to fetch nodes by key |
```
Comment (goddammit) [inline doc/modules/mod_pubsub.md]: Time to fetch nodes by key.

File: doc/modules/mod_pubsub.md [hunk 7]
```diff
@@ -109,7 +101,7 @@ Requires module `mod_caps` to be enabled.
 #### `<<"dag">>`
 
 Implementation of [XEP-0248 (PubSub Collection Nodes)](https://xmpp.org/extensions/xep-0248.html).
-Every node takes a place in a tree and is either a collection node (and have only sub-nodes) or a leaf node (contains only items).
+Every node takes a place in a collection and is either a collection node (and have only sub-nodes) or a leaf node (contains only items).
```
Comment (goddammit) [inline doc/modules/mod_pubsub.md]: Every node takes a place in a collection and becomes either a collection node (and have only sub-nodes) or a leaf node (contains only items).

File: doc/modules/mod_pubsub.md [hunk 8]
```diff
@@ -132,4 +124,18 @@ If you'd like to learn more about metrics in MongooseIM, please visit the [Mongo
 | `[global, backends, mod_pubsub_db, get_states_by_bare]` | histogram | Time to fetch nodes' states for bare JID. |
 | `[global, backends, mod_pubsub_db, get_states_by_full]` | histogram | Time to fetch nodes' states for full JID. |
 | `[global, backends, mod_pubsub_db, get_own_nodes_states]` | histogram | Time to fetch state data for user's nodes. |
-
+| `[global, backends, mod_pubsub_db, create_node, ]` | histogram | Time to set the owner after node is created. |
```
Comment (goddammit) [inline doc/modules/mod_pubsub.md]: Time to set the owner after a node is created.

File: doc/modules/mod_pubsub.md [hunk 9]
```diff
@@ -65,16 +65,8 @@ Only one nodetree can be used per host and is shared by all node plugins.
...
-#### `<<"virtual">>`
-
-Does not store nodes in a database.
-This saves resources on systems when the number of nodes is large.
-When using the `<<"virtual">>` nodetree, you can only enable these node plugins: `[<<“flat”>>,<<“pep”>>]` or `[<<“flat”>>]`; any other plugin configuration will not work.
-Also, all nodes will have the defaut configuration, and this can not be changed.
-
-Using the `<<“virtual”>>` nodetree requires the pubsub module to be started with a clean database, it will not work if you used the default `<<“tree”>>` nodetree before.
+Stores nodes in a database (either `mnesia` or `rdbms`).
+Allows to organise nodes in trees if `hometree` plugin is used.
```
Comment (goddammit) [inline doc/modules/mod_pubsub.md]: Allows to organise nodes in trees if the `hometree` plugin is used.

**Pull request merged**